var/spack/repos/builtin/packages/xsdk/package.py

```diff
@@ -111,8 +150,27 @@ class Xsdk(BundlePackage):
     # these are type='build' dependencies, but spack reports a conflict anyway.
     # This will be fixed once the new concretizer becomes available
     # (says @adamjstewart)
-    depends_on('phist@develop kernel_lib=tpetra ~fortran ~scamac ~openmp ~host', when='@develop +phist')
-    depends_on('phist@1.7.5 kernel_lib=tpetra ~fortran ~scamac ~openmp ~host', when='@0.4.0 +phist')
+
+    depends_on('phist kernel_lib=tpetra', when='+trilinos +phist')
+    depends_on('phist kernel_lib=petsc', when='~trilinos +phist')
+    depends_on('phist@develop ~fortran ~scamac ~openmp ~host', when='@develop +phist')
+    depends_on('phist@1.8.0 ~fortran ~scamac ~openmp ~host', when='@0.5.0 +phist')
+    depends_on('phist@1.7.5 ~fortran ~scamac ~openmp ~host', when='@0.4.0 +phist')
+
+    depends_on('ginkgo@develop ~openmp', when='@develop +ginkgo')
+    depends_on('ginkgo@develop ~openmp+cuda', when='@develop +ginkgo +cuda')
+    depends_on('ginkgo@1.1.0 ~openmp', when='@0.5.0 +ginkgo')
+    depends_on('ginkgo@1.1.0 ~openmp+cuda', when='@0.5.0 +cuda +ginkgo')
+
+    depends_on('py-libensemble@develop+petsc4py', when='@develop +libensemble')
+    depends_on('py-libensemble@0.5.2+petsc4py', when='@0.5.0 +libensemble')
```

Comment:
These might be `type=('build', 'run')` dependencies.

Reply:
Hm - `py-libensemble` is another numerical package and not really a build dependency

Reply:
Now that I think about it, it really doesn't matter what dependency type most of these packages are, as there is no code in `xsdk` to link to. In that case, this can just be `type='run'` since it needs to be present in `PYTHONPATH` at run-time.

Reply:
ok - pushed